src/main/java/org/folio/rest/support/EffectiveCallNumberComponentsUtil.java

```diff
@@ -0,0 +1,51 @@
+package org.folio.rest.support;
+
+import org.apache.commons.lang.StringUtils;
+import org.folio.rest.jaxrs.model.EffectiveCallNumberComponents;
+import org.folio.rest.jaxrs.model.HoldingsRecord;
+import org.folio.rest.jaxrs.model.Item;
+
+/**
+ * Utility methods for Effective Call Number Components object.
+ */
+public final class EffectiveCallNumberComponentsUtil {
+
+  private EffectiveCallNumberComponentsUtil() {
+    throw new UnsupportedOperationException("Do not instantiate");
+  }
+
+  public static EffectiveCallNumberComponents buildComponents(HoldingsRecord holdings, Item item) {
+    return updateComponents(new EffectiveCallNumberComponents(), holdings, item);
+  }
+
+  private static EffectiveCallNumberComponents updateComponents(
+    EffectiveCallNumberComponents components, HoldingsRecord holdings, Item item) {
+
+    String updatedCallNumber = null;
+    if (StringUtils.isNotBlank(item.getItemLevelCallNumber())) {
```

Comment:
I find it easier to read statements like this with static imports, as I find the utility class names distracting (and long for extension methods, that at first I really disliked).

```suggestion
    if (isNotBlank(item.getItemLevelCallNumber())) {
```